File: front-project/src/app/models/post.ts
import { Reply } from './reply';

export interface Post {
  id: number;
  nick: string;
  categoryId: number;
  body: string;
  createdAt: Date | string;
  verified: boolean;
  reply: Reply[];
}

export interface PostPost {
  nick: string;
  categoryId: number;
  body: string;
  createdAt: Date | string;
  verified: boolean;
  reply: Reply[] | null;
}

export interface UpdatePost {
  nick: string;
  categoryId: number;
  body: string;
  verified: boolean;
}

File: BackProject/Models/Post.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace BackProject.Models
{
    public class Post
    {
        public int Id { get; set; }
        [Required]
        public String Nick { get; set; }
        [Required]
        public int CategoryId { get; set; }
        [Required]
        public String Body { get; set; }
        public DateTime CreatedAt { get; set; }
        public Boolean Verified { get; set; }
        public List<Reply> Reply { get; set; }
    }
}